docs/test_env_setup.rst

```diff
@@ -15,5 +17,116 @@ For this reason, all post installation and test setup configuration steps
 are automated via ansible playbooks and stored in a separate `usmqe-setup
 repository`_. You need to deploy test machines using playbooks from there.
 
+
+Ansible Roles for Tendrl Setup
+==============================
+
+Right now, we maintain ansible roles and playbook which automates production
+like installation of all Tendrl components. `Tendrl project wide
+documentation`_ is followed, but most details are based on documentation or
+README files of each component.
+
+For each Tendrl component, there is a role in `usmqe-setup repository`_. For
+example, there are:
+
+* `tendrl-common`_
+* `tendrl-node-agent`_
+* `tendrl-api`_
+
+Note: the list presented doesn't include all roles. You would need to check
+``roles`` directory in the repository to see current full list of roles.
+
+Every ansible role of particular tendrl component has the following tasks
+structure::
+
+    tasks/
+    ├── binary.yml
+    ├── main.yml
+    └── source.ym
+
+Where ``main.yml`` looks like this (the example comes from ``tendrl-common``)::
+
+    #
+    # installation
+    #
+    
+    - include: source.yml
+      when: install_from == "source"
+    
+    - include: binary.yml
+      when: install_from == "packages"
+    
+    #
+    # post installation configuration
+    #
+    
+    - name: Specify etcd_connection in tendrl.conf
+      ini_file:
+        dest=/etc/tendrl/tendrl.conf
+        section=common
+        option=etcd_connection
+        value="{{ groups['usm_server'][0] }}"
+
+As you can see, there are 2 additional yaml files, one for installation from
+rpm packages and other for installation from source code. The default value of
```

Comment:
Do we expect to have also deb packages for Debian/Ubuntu? I know it will require lot's of changes, but still if it is expected to support also Debian like distributions I would remove the _rpm_ reference here (so it will stay as `...installation from packages...`).
The same apply also for the next paragraph.